### ISSUE TITLE
Billing MT, Site taking longer to open "Site Edit Page"

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#199]
In GitLab by @smasih on Jan 16, 2017, 05:26

From: <LGPudles@aol.com>

Date: 16-Jan-2017 03:28

Subject: Re: Edit Site Settings Issue

To: <coribartlett@gmail.com>, <sabhelp@answernet.com>

|

I just had the same thing happen to me

|
 
In a message dated 1/15/2017 1:41:48 P.M. Eastern Standard Time, coribartlett@gmail.com writes:

|

 Hi I've tried to edit Site settings for Billings & a couple of other sites yesterday & today it just clocks and never allows access. 

|

Cori Bartlett

Cori.bartlett@answernet.com

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 16, 2017, 05:32

@abakshi @tim.traylor @kdjstudios 

Hi all,


First I checked on Ldap , if the code was returning user ldap level properly. Well it was, there was no issue with ldap.

It just Billing MT has 524 additional fees plus activities, holidays etc so site loading is taking time.

However I made few changes which is showing slight improvement in site page loading.

If still we find the wait longer, we can make changes on Edit Site page and load the additional fees or activities on the page only when specifically requested through a button say Load items (through AJAX). 

Please check and let us know.

**Comment by kdjstudios** [comment]
In GitLab by @rchauhan on Jan 19, 2017, 08:44

* Check code change on Git that might be causing problems. 
* Deeper analysis of the code. 

We have gone from 3 minutes to under a minute. Needs to be 10 seconds on average.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jan 19, 2017, 08:46

Additional Javascript to disable additional activities might be taking more time. Need to analyse this in more detail.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jan 20, 2017, 24:09

@smasih Please analyse and update (*Estimate: 1 hr*)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 31, 2017, 09:31

@abakshi @smasih 

I just reloaded the Billings sites edit page on Staging and Production. Each of them took roughly 1 minute to open. Was there any updates on this?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 7, 2017, 07:51

@kdjstudios 

We were successful optimizing the site edit page, but still under testing.

Will let you know once its done and release it on Feb 18, 2017.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 7, 2017, 08:53

@smasih 

Thanks for the update. I will update Cori and the others.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Feb 15, 2017, 24:42

Paused for now to complete the other higher priority items.  

Estimate: ~8 hours

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 16, 2017, 05:47

Milestone removed

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Mar 24, 2017, 02:48

@kdjstudios @tim.traylor 

Just checked with team and it seems that since this is a low priority item as compared to other open items, we may need to revisit this for mid/end of April release.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 24, 2017, 09:49

@abakshi 

Thank you for the update. I agreed lets plan for a revisit at the mid/end of April.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 4, 2017, 02:24

Revisiting this ticket... current priority is #305 
Can pick after this release

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jun 15, 2017, 04:49

Require 2 days effort. Still working on #305 Let's review this along with merge #117

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 29, 2017, 01:31

Milestone removed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 29, 2017, 11:02

@abakshi @smasih 

Gary Wrote: http://www.servicedesk.answernet.com/profiles/ticket/2017-08-29-54651/conversation

I am trying to get into billings settings and its taking a very long time. 

Kyle-

Communicating with Gary via IM, he stated it was taking around five minutes to load.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 30, 2017, 08:04

@kdjstudios @abakshi 

As soon as we are done with aging, new site and merge testing we will switch back to this issue. We know it's there and were hoping to resolve it before we make merge code live.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 30, 2017, 08:15

@smasih @tim.traylor 

Okay thanks, we also want to make sure these changes effect all sites, not just Billings that seems to be the main site that takes the longest.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 04:57

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 10, 2018, 04:53

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 22, 2018, 02:40

@kdjstudios 

Optimized site load, but however in Billing MT there are `537` additional fees and those take time.

As an optimization, we are loading `site additional` fees in parallel.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 22, 2018, 10:16

@smasih 

Thank you. Were these changes done on Merge, right?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 22, 2018, 22:20

Marking Primary for #636

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 22, 2018, 23:28

> Thank you. Were these changes done on Merge, right?

Yes, these changes were done on merge.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 23, 2018, 08:14

@smasih 

I think this is great, I see that the page will load (under 10 seconds) and then display a loading circle in the additional fees, until those load too (about a minute). 

I do have a question for clarification. I thought it was mentioned at some point to load like 10 records, and then display a button to load more. However I do not recall what was decided with that or why we went this way instead.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 24, 2018, 24:04

@kdjstudios 

As far as I remember the discussion it was either to display on scroll or give a button to display but we never finalized the decision. I tried both scroll and button and found this way to be more user-friendly and less confusing. However, if this does not suffice we can try something else.

@tim.traylor 

This was just a small modification, we are not using the framework and db to its full potential. Once we have the reports project completed we can do more modifications to make the load time 2 - 3 seconds.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2018, 07:50

Placing in a pause status, as Sumeet states there is still more to do once the reports are completed.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 24, 2018, 23:27

I suggest that we should create another ticket once approved from @tim.traylor 

Because in order to make it faster we need to change the whole backend architecture of the Site edit page.

So if current improvement is satisfactory for the time being, in that case, we can close this one but's let's keep it in the discussion until unless we decide to go further.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 25, 2018, 09:35

Agreed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 14, 2018, 10:52

@smasih @tim.traylor 

New ticket from Gary: http://www.servicedesk.answernet.com/profiles/ticket/2018-02-14-81917/conversation

> It always takes a long time to open the edit site screen.  Any chance we can fix that?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 15, 2018, 03:55

@kdjstudios 

Was the improvement on merge not satisfactory ..? or maybe Garry was referring to the production server.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 15, 2018, 07:58

Gary is only working on Production, so it those fixes were not yet released that would most likely be why. As Production would still be slow then right?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 15, 2018, 23:49

anyhow, once merge is released this saturday it would be automatically taken care of.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 16, 2018, 08:28

Perfect. :-)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 30, 2018, 13:30

Marking for complete. Merge was released.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2018, 13:09

closed
